src/test/python/pythonparser/test_pythonparser_3.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) Aniskov N.
+
+import os
+from xml.etree import ElementTree as ET
+from typing import *
+
+import pytest
+
+from src.main.python.pythonparser.pythonparser_3 import *
+from src.main.util.const import RESOURCE_FOLDER_PATH
+
+
+class TestPythonParser3:
+
+    _input_dir = os.path.join(RESOURCE_FOLDER_PATH, 'pythonparser/pythonparser_3')
+
+    test_cases_dirs = \
+        [
+            os.path.join(_input_dir, 'case_small_code_snippets'),
+
+            os.path.join(_input_dir, 'case_big_code_snippets'),
+
+            os.path.join(_input_dir, 'case_classes'),
+
+            os.path.join(_input_dir, 'case_empty_file'),
+
+            pytest.param(os.path.join(_input_dir, 'case_fstrings'),
+                         marks=pytest.mark.skip(reason='fstrings are not supported yet')),
+
+            pytest.param(os.path.join(_input_dir, 'case_python_3_8'),
+                         marks=pytest.mark.skip(reason='python 3.8 not supported yet'))
+        ]
+
+    # Helper functions
+
+    @staticmethod
+    def get_xml_tree(path_to_py_file: str) -> ET.Element:
+        parsed_json = parse_file(path_to_py_file)
+        xml_str = json2xml(parsed_json)
+        root = ET.fromstring(xml_str)
+        return root
+
+    @staticmethod
+    def count_token_occurrence(tree: ET.Element, tokens: Iterable[str]) -> Dict[str, int]:
+        result = {token: len(tree.findall('.//' + token))
+                  for token in tokens}
+        return result
+
+    # Tests
+
+    @pytest.mark.parametrize(
+        'cases_dirs',
+        test_cases_dirs
+    )
+    def test_produce_valid_xml(self, cases_dirs: str) -> None:
+        """
+        :arg: list of input dirs (pytest parametrize)
+        Checks if parser producing valid xml file
+        """
+
+        for entry in os.scandir(cases_dirs):
+            # print(f'Current entry is: {entry.path}') TODO: remove or add logger instead
+            parsed_json = parse_file(entry.path)
+            xml_str = json2xml(parsed_json)
+            try:
+                _ = ET.fromstring(xml_str)
+            except ET.ParseError as e:
+                pytest.fail(f'test file: {entry.path}, ET.ParseError msg: {e}')
+
+    @pytest.mark.parametrize(
+        'test_input_dirs',
+        [
+            os.path.join(_input_dir, 'case_non_valid_python_code')
+        ]
+    )
+    def test_non_valid_python_code(self, test_input_dirs: str) -> None:
+        """
+        Checks behaviour on non valid python code inputs.
+        Expected behaviour: SyntaxError is thrown during parsing
+        """
+        for entry in os.scandir(test_input_dirs):
+            # print(f'Current entry is: {entry.path}') TODO: remove or add logger instead
+            with pytest.raises(SyntaxError) as exc_info:
```

Comment:
You don't use `exc_info`. It would be better to delete it: `with pytest.raises(SyntaxError):`